node/api/renter.go

```diff
@@ -319,6 +320,44 @@ func (api *API) renterContractsHandler(w http.ResponseWriter, _ *http.Request, _
 	})
 }
 
+// renterClearDownloadsHandler handles the API call to request to clear the download queue.
+func (api *API) renterClearDownloadsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var startInt, endInt int64
+	var err error
+
+	startStr, endStr := req.FormValue("start"), req.FormValue("end")
+
+	if startStr != "" {
+		startInt, err = strconv.ParseInt(startStr, 10, 64)
+		if err != nil {
+			WriteError(w, Error{"parsing integer value for parameter `start` failed: " + err.Error()}, http.StatusBadRequest)
+			return
+		}
+	}
+	startTime := time.Unix(0, startInt)
+
+	if endStr != "" {
+		endInt, err = strconv.ParseInt(endStr, 10, 64)
+		if err != nil {
+			WriteError(w, Error{"parsing integer value for parameter `end` failed: " + err.Error()}, http.StatusBadRequest)
+			return
+		}
+	}
+	endTime := time.Unix(0, endInt)
+
+	if startTime.Before(endTime) && !startTime.Equal(time.Unix(0, 0)) {
+		WriteError(w, Error{"start time can not be before end time"}, http.StatusBadRequest)
```

Comment:
I would move this check inside the renter function, since we'll want to catch it in Go programs as well

Reply:
:+1: